File: src/HRMOptimus.Application/Employee/Query/Employees/EmployeeVm.cs
using HRMOptimus.Domain.Enums;
using System;
using System.Collections.Generic;

namespace HRMOptimus.Application.Employee.Query.Employees
{
    public record EmployeeVm(int Id, string FirstName, string LastName, string FullName, Gender Gender,
        DateTime? BirthDate, string Email, string PhoneNumber, string ContractName, List<string> roles);
}

File: client/src/shared/vm/employee.vm.ts
import { Gender } from '../enums/gender';
export class EmployeeVm {
  id: number | undefined;
  employeeId: number | undefined;
  firstName: string | undefined;
  lastName: string | undefined;
  fullName: string | undefined;
  gender: Gender | undefined;
  birthDate: string | undefined;
  email: string | undefined;
  phoneNumber: string | undefined;
  items: any | undefined;
  address: {
    street: string | undefined;
    buildingNumber: string | undefined;
    houseNumber: string | undefined;
    zipCode: string | undefined;
    city: string | undefined;
    country: string | undefined;
  };
  contract: {
    contractName: string | undefined;
    contractType: number | undefined;
    leaveDays: number | undefined;
  };
  roles: string[] | undefined;
}
